gui/src/main/java/cern/jarrace/inspector/gui/rest/ContainerService.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * © Copyright 2016 CERN. This software is distributed under the terms of the Apache License Version 2.0, copied
+ * verbatim in the file “COPYING”. In applying this licence, CERN does not waive the privileges and immunities granted
+ * to it by virtue of its status as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package cern.jarrace.inspector.gui.rest;
+
+import cern.jarrace.commons.domain.AgentContainer;
+import retrofit2.Call;
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+import java.util.List;
+
+/**
+ * A service that can access a remote endpoint with methods related to {@link AgentContainer}s.
+ */
+public interface ContainerService {
```

Comment:
Regarding domain objects I might be inclined to agree (and even so we might have different domain objects on both sides depending on how much you want to deprive the user of internals), now regarding the implementation of the consumption and provisioning of services I disagree. In each layer the information that you need to pass is slighly different because they have diferent concerns. Just because we have control over both sides does not make it mandatory to use a common technology for both in my opinion.